bitcoin/src/blockdata/witness.rs

```diff
@@ -206,6 +189,66 @@ crate::internal_macros::define_extension_trait! {
     }
 }
 
+enum P2TrSpend<'a> {
+    Key {
+        // signature: &'a [u8],
```

Comment:
In 95ff586259191022255e2ce6808cc6809784bb26:

How come this (and its constructor below) are commented out?

Reply:
It's not used by anything (yet) so it'd just blow up the variant size and produce compiler warnings. However since I originally wrote it with the field I decided to keep it around commented since I suspect the likelihood of something using it in the future is quite high.

Reply:
I don't like the code comments left in there unless:

1. You have something ether in mind or queued up that adds them
2. We add more comments describing why the field (a) is not there (b) is left in but commented out

As it is it took me a while to workout what the commented out bits meant.

Reply:
I think we might add `taproot_key_spend_signature` method or a similar one one day but also it's possible we will just make the enum public. So that's what I have in mind. It also clarifies that the field is in the witness but not used by our code. I can add the explaining comment though.

Reply:
I'd be fine with just an explaining comment. I think we will want to add `taproot_key_spend_signature` in a followup.

Reply:
Alright, I'll add it.